### PR TITLE
refactor: deprecate gateway mode

### LINKED ISSUE #2972: Deprecate gateway mode
# Task

There is an undocumented gateway mode (`gateway-mode` config variable) which disables certain endpoints. With the Restricted API and the [gateway-proxy](https://github.com/ethersphere/gateway-proxy) released the gateway mode is no longer need to be in Bee. Once the Restricted API is no longer experimental and the gateway-proxy is documented the gateway mode can be removed. In the meanwhile the deprecation process can be started.

## Acceptance criterea
The `gateway-mode` config variable is removed and all the code that forbids endpoints from being accessed is also removed.

### PR DESCRIPTION
**Pull request by swarmHaseeb**
refactor: deprecate gateway mode
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [x] My change requires a documentation update and I have done it
- [x] I have filled out the description and linked the related issues

### Description

Deprecate gateway mode and removing all related code and update documentation
#### Motivation and context (Optional)

Api refactoring 
### Related Issue (Optional)

closes #2972 
### Screenshots (if appropriate):

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3265)

### CODE REVIEW

File: packaging/bee.yaml [hunk 1]
```diff
@@ -33,7 +33,6 @@ debug-api-addr: 127.0.0.1:1635
 ## enable debug HTTP API
 debug-api-enable: true
 ## disable a set of sensitive features in the api
-# gateway-mode: false
```
Comment (vladopajic) [inline packaging/bee.yaml:36]: Comment should be also removed.

File: packaging/docker/env [hunk 2]
```diff
@@ -40,7 +40,6 @@ BEE_CLEF_SIGNER_ENABLE=true
 ## enable debug HTTP API
 # BEE_DEBUG_API_ENABLE=false
 ## disable a set of sensitive features in the api
```
Comment (vladopajic) [inline packaging/docker/env:42]: Comment should be also removed.

File: packaging/homebrew-amd64/bee.yaml [hunk 3]
```diff
@@ -33,7 +33,6 @@ debug-api-addr: 127.0.0.1:1635
 ## enable debug HTTP API
 debug-api-enable: true
 ## disable a set of sensitive features in the api
-# gateway-mode: false
```
Comment (vladopajic) [inline packaging/homebrew-amd64/bee.yaml:36]: Comment should be also removed.

File: packaging/homebrew-arm64/bee.yaml [hunk 4]
```diff
@@ -33,7 +33,6 @@ debug-api-addr: 127.0.0.1:1635
 ## enable debug HTTP API
 debug-api-enable: true
 ## disable a set of sensitive features in the api
-# gateway-mode: false
```
Comment (vladopajic) [inline packaging/homebrew-arm64/bee.yaml:36]: Comment should be also removed.

File: packaging/scoop/bee.yaml [hunk 5]
```diff
@@ -23,7 +23,6 @@ data-dir: ./data
 ## enable debug HTTP API
 # debug-api-enable: false
 ## disable a set of sensitive features in the api
-# gateway-mode: false
```
Comment (vladopajic) [inline packaging/scoop/bee.yaml:26]: Comment should be also removed.

**Review by swarmHaseeb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3265)* status: 0 of 17 files reviewed, 1 unresolved discussion (waiting on @istae, @mrekucci, @umerm-work, and @vladopajic)

___
*[`packaging/docker/env` line 42 at r1](https://reviewable.io/reviews/ethersphere/bee/3265#-NBWOgFAK9Ag1upjPjBR-r1-42:-NBgZfVNDkir4U6JwcAc:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/d5fc952d1cf1fd47e8f2dab0f62408e1af10eb4e/packaging/docker/env#L42)):*
<details><summary><i>Previously, vladopajic (Vlado Pajić) wrote…</i></summary><blockquote>

Comment should be also removed.
</blockquote></details>

Done.

**Review by vladopajic — COMMENTED** [review]
LGTM

**Review by notanatol — APPROVED** [review]

**Review by vladopajic — APPROVED** [review]

**Review by mrekucci — CHANGES_REQUESTED** [review]
The gateway mode is marked as deprecated, yet it is removed from the code. If a feature or API is marked as deprecated, it should work until the next release and then be removed completely. Either mark it as deprecated in the Open API specification and create a subsequent PR (where it will be removed entirely) that will be included in the release following the release in which it was marked as deprecated. Or remove it completely also from the Open API specification.

**Review by mrekucci — CHANGES_REQUESTED** [review]
Since the HTTP response set has changed, the version of the Open API specification should also change.

File: openapi/SwarmDebug.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.3
 info:
-  version: 2.1.0
+  version: 2.1.1
```
Comment (mrekucci) [inline openapi/SwarmDebug.yaml:3]: There has been no change in the debug API, so there is no need to change the version.

**Review by mrekucci — CHANGES_REQUESTED** [review]
We try to follow the [semver](https://semver.org/) approach, so in this case, there are no bug fixes or incompatible API changes, so the minor version should be incremented.

File: pkg/api/router.go [hunk 7]
```diff
@@ -258,12 +255,10 @@ func (s *Service) mountAPI() {
 	)
 
 	handle("/pss/subscribe/{topic}", web.ChainHandlers(
-		s.gatewayModeForbidEndpointHandler,
```
Comment (istae) [inline pkg/api/router.go:261]: Here we remove the gateway forbidden middleware which blocks users from requesting the tags of all uploads, including others'. Has this been thoroughly discussed ?

**Review by istae — CHANGES_REQUESTED** [review]

**Review by istae — APPROVED** [review]

**Review by mrekucci — APPROVED** [review]

**Pull request merged**